src/sidebar/services/frame-sync.js

```diff
@@ -7,12 +7,14 @@ import * as metadata from '../util/annotation-metadata';
 import { watch } from '../util/watch';
 
 /**
- * @typedef FrameInfo
+ * @typedef {import('../../types/annotator').DocumentMetadata} DocumentMetadata
+ */
+
+/**
+ * @typedef Frame
+ * @property {string|null} id
```

Comment:
I think it would be useful to add a comment explaining what it means for a frame to have an ID vs no ID. That information is currently buried away in a comment: https://github.com/hypothesis/client/blob/240cb4781d43ffa19e2ebac9ec7f3b683d2ff38c/src/sidebar/store/modules/frames.js#L99